src/thyra/ThyraProvider.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EAccountDeletionResponse,
+  IAccountDeletionRequest,
+  IAccountDeletionResponse,
+} from '../provider/AccountDeletion';
+import {
+  IAccountImportRequest,
+  IAccountImportResponse,
+} from '../provider/AccountImport';
+import { IProvider } from '../provider/IProvider';
+import {
+  JsonRpcResponseData,
+  deleteRequest,
+  getRequest,
+} from './RequestHandler';
+import { ThyraAccount } from './ThyraAccount';
+import { IAccount } from '../account/IAccount';
+
+export const THYRA_ACCOUNTS_URL =
+  'https://my.massa/thyra/plugin/massalabs/wallet/rest/wallet';
+
+export const THYRA_IMPORT_ACCOUNTS_URL = `${THYRA_ACCOUNTS_URL}/import/`;
+
+export const THYRA_PROVIDER_NAME = 'THYRA';
+
+export interface IThyraWallet {
+  address: string;
+  nickname: string;
+  keyPair: {
+    nonce: string;
+    privateKey: string;
+    publicKey: string;
+    salt: string;
+  };
+}
+
+export class ThyraProvider implements IProvider {
+  private providerName: string;
+
+  public constructor() {
+    this.providerName = THYRA_PROVIDER_NAME;
+  }
+
+  public name(): string {
+    return this.providerName;
+  }
+
+  public async accounts(): Promise<IAccount[]> {
+    let thyraAccountsResponse: JsonRpcResponseData<Array<IThyraWallet>> = null;
+    try {
+      thyraAccountsResponse = await getRequest<Array<IThyraWallet>>(
+        THYRA_ACCOUNTS_URL,
+      );
+    } catch (ex) {
+      console.error(`Thyra accounts retrieval error`);
+      throw ex;
+    }
+    if (thyraAccountsResponse.isError || thyraAccountsResponse.error) {
+      throw thyraAccountsResponse.error.message;
+    }
+    const thyraAccounts: IThyraWallet[] = thyraAccountsResponse.result;
+    let accounts: ThyraAccount[] = [];
+    for (const thyraAccount of thyraAccounts) {
+      const accInstance = new ThyraAccount(
+        { address: thyraAccount.address, name: thyraAccount.nickname },
+        this.providerName,
+      );
+      accounts.push(accInstance);
+    }
+    return accounts;
+  }
+
+  public async importAccount(
+    publicKey: string,
+    privateKey: string,
+  ): Promise<IAccountImportResponse> {
+    // TODO: cannot import as it goes over the popup dialog (need special url for that)
+    return null;
+  }
+
+  public async deleteAccount(
+    address: string,
```

Comment:
no blocking

If my app knows the nickname, it's unoptimal to pass the address because this function will retrieve the nickame. Is it planed to add a function deteleAccountByNickame(nickname: string)?

Reply:
@gregLibert I believe @Thykof 's comment is justified - the question is : shall thyra expose further rest api methods such as delete walletByAddress or should we rather change the focus of the standard that wallet-provider implements to deleting wallets (=accounts) according to a name rather than address as it is now.

Reply:
In the standard, the wallet nickname is optional where the address is mandatory.
Therefore, the code for this library will stay deleteAccount using an address, unless we update the standard on this subject which is not planed.

If on Thyra side all accounts have nickname, this library can keep a way to translate an address to a nickname and use the nickname when excahnging with Thyra.